File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/sales/customer/CustomerResponseDTO.java
package com.example.fieldsync_inventory_api.dto.sales.customer;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.example.fieldsync_inventory_api.dto.sales.customer_type.CustomerTypeResponseDTO;

import java.time.Instant;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CustomerResponseDTO {
    private Integer id;
    private String fullName;
    private String email;
    private String phone;
    private String address;
    private CustomerTypeResponseDTO customerType;
    private Instant createdAt;
    private Instant updatedAt;
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/customer/CustomerResponseDto.kt
package com.example.fieldsync_inventory_app.data.remote.dto.customer

// Response DTO (Updated to include the embedded CustomerType object)
data class CustomerResponseDto(
    val id: Int,
    val fullName: String,
    val email: String?,
    val phone: String?,
    val address: String?,
    val customerType: EmbeddedCustomerTypeDto,
    val createdAt: String,
    val updatedAt: String,
    val deletedAt: String?
)